test/services/user-group/services/courseModelService.test.js

```diff
@@ -0,0 +1,79 @@
+const { courseModel } = require('../../../../src/services/user-group/model');
+
+describe('course model service', () => {
+	let server;
+
+	before((done) => {
+		server = app.listen(0, done);
+	});
+
+	after(async () => {
+		await testObjects.cleanup();
+		await server.close();
+	});
+
+
+	it('CREATE a course on internal call', async () => {
+		const { _id: schoolId } = await testObjects.createTestSchool({});
+		const { _id: teacherId } = await testObjects.createTestUser({ roles: ['teacher'], schoolId });
+		const result = await app.service('courseModel').create({
+			name: 'testcourse',
+			schoolId,
+			teacherIds: [teacherId],
+			substitutionIds: [],
+			classIds: [],
+			userIds: [],
+		});
+		expect(result).to.not.be.undefined;
+		expect(result).to.haveOwnProperty('_id');
+		expect(result.name).to.eq('testcourse');
+		expect(result.schoolId.toString()).to.eq(schoolId.toString());
+		// make sure course was saved to db
+		const dbResult = await courseModel.findById(result._id).lean().exec();
+		expect(dbResult).to.not.be.undefined;
+		expect(dbResult).to.haveOwnProperty('_id');
+		expect(dbResult.name).to.eq('testcourse');
+		expect(dbResult.schoolId.toString()).to.eq(schoolId.toString());
+	});
+
+	it('GET a course on internal call', async () => {
+		const { _id: schoolId } = await testObjects.createTestSchool({});
+		const { _id: teacherId } = await testObjects.createTestUser({ roles: ['teacher'], schoolId });
+		const course = await testObjects.createTestCourse({ schoolId, teacherIds: [teacherId] });
+		const result = await app.service('courseModel').get(course._id);
+		expect(result).to.not.be.undefined;
+		expect(result).to.haveOwnProperty('_id');
+		expect(result).to.haveOwnProperty('name');
+		expect(result.schoolId.toString()).to.eq(schoolId.toString());
+	});
+
+	it('FIND courses on internal call', async () => {
+		const { _id: schoolId } = await testObjects.createTestSchool({});
+		const { _id: teacherId } = await testObjects.createTestUser({ roles: ['teacher'], schoolId });
+		const course = await testObjects.createTestCourse({ schoolId, teacherIds: [teacherId] });
+		const result = await app.service('courseModel').find({ query: { _id: course._id } });
+		expect(result.data).to.not.be.undefined;
+		expect(result.total).to.equal(1);
+		expect(result.data[0]).to.haveOwnProperty('_id');
+		expect(result.data[0]).to.haveOwnProperty('name');
+		expect(result.data[0].schoolId.toString()).to.eq(schoolId.toString());
+	});
+
+	it('external call is blocked', async () => {
+		const { _id: schoolId } = await testObjects.createTestSchool({});
+		const teacher = await testObjects.createTestUser({ roles: ['teacher'], schoolId });
+		const course = await testObjects.createTestCourse({ schoolId, teacherIds: [teacher._id] });
+		const params = await testObjects.generateRequestParamsFromUser(teacher);
+		try {
+			const result = await app.service('courseModel').get(course._id, params);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: ['result' is assigned a value but never used.](https://app.codacy.com/manual/schul-cloud/schulcloud-server/pullRequest?prid=5271145)